From: Marcus Barrow <mbarrow@redhat.com>
Date: Thu, 20 Dec 2007 13:08:15 -0500
Subject: [scsi] qla2xxx: fix bad nvram kernel panic
Message-id: 20071220180815.15892.29755.sendpatchset@shell.boston.redhat.com
O-Subject: [Bug 367201][QLogic][RHEL 5.2 Patch]qla2xxx: regression fix bad nvram kernel panic
Bugzilla: 367201

This patch fixes a regression in 5.1 caused by not configuring
boards with corrupted NVRAMs. This regression causes a kernel panic().

    [SCSI] qla2xxx: Error-out during probe() if we're unable to complete HBA initialization.

    Remove a stale check against ha->device_flags
    (DFLG_NO_CABLE) as topology scanning is performed within the
    DPC-thread context.

    From upstream commit: 765140bf22adfaecd7077ce10cd6198ef872094c

Acked-by: Prarit Bhargava <prarit@redhat.com>
Acked-by: Chip Coldwell <coldwell@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index ed5a90d..3c794c7 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -1729,9 +1729,7 @@ qla2x00_probe_one(struct pci_dev *pdev, const struct pci_device_id *id)
 		goto probe_failed;
 	}
 
-	if (qla2x00_initialize_adapter(ha) &&
-	    !(ha->device_flags & DFLG_NO_CABLE)) {
-
+	if (qla2x00_initialize_adapter(ha)) {
 		qla_printk(KERN_WARNING, ha,
 		    "Failed to initialize adapter\n");
 
